File: spring-boot-template/src/main/java/tui/meta/challenge/quotes/service/QuoteService.java
package tui.meta.challenge.quotes.service;

import tui.meta.challenge.quotes.controller.model.QuoteDto;

import java.util.List;

public interface QuoteService {
    QuoteDto getSingleQuote(String id);

    List<QuoteDto> getQuotesByAuthor(String author);

    List<QuoteDto> getAllQuotes();
}

File: micronaut-template/src/main/kotlin/tui/meta/challenge/service/QuoteService.kt
package tui.meta.challenge.service

import tui.meta.challenge.controller.model.QuoteDto

interface QuoteService {
    fun getQuoteById(id: String): QuoteDto
    fun getAllQuotes(): List<QuoteDto>
    fun getQuotesByAuthor(author: String): List<QuoteDto>
}